scripts/superdesk/editor/spellcheck/spellcheck.js

```diff
@@ -77,6 +77,101 @@ function SpellcheckService($q, api, dictionaries) {
+                if (item === abbrevWord) {
+                    _abbrevArr.push(abbrevWord);
+                }
+            });
+        });
+        var _abbreviationString = _abbrevArr.join('|');
+
+        // Prepare non sentence words
+        var match, wordIndex;
+        var nonSentenceWords = [];
+
+        // Words that follows abbreviations in content
+        var _reNonSentenceWords = '\\s+(?:' + _abbreviationString + ')(\\s+\\w+)';
+        var reNonSentenceWords = new RegExp(_reNonSentenceWords, 'g');
+        while ((match = reNonSentenceWords.exec(textContent)) != null) {
+            wordIndex = match.index + match[0].indexOf(_.trim(match[1]));
+            nonSentenceWords.push({
+                word: _.trim(match[1]),
+                index: currentOffset + wordIndex
+            });
+        }
+
+        return nonSentenceWords;
+    }
+
+    function testExpression(textContent, currentOffset, abbreviationList) {
+        var reSentenceWords = /(?:^|(?:[.|?|!]\s+))(\w+)/g; // words followed by .|?|!
+        var match, wordIndex;
+        var sentenceWords = [];
+        while ((match = reSentenceWords.exec(textContent)) != null) {
+            wordIndex = match.index + match[0].indexOf(match[1]);
+            sentenceWords.push({
+                word: match[1],
+                index: currentOffset + wordIndex
+            });
+        }
+
+        // Exclude the words form capitalization if it follows by abbreviation.
+        var nonSentenceWords = getNonSentenceWords(textContent, currentOffset, abbreviationList);
+
+        _.forEach(nonSentenceWords, function(nonSentenceWord) {
+            _.remove(sentenceWords, {index: nonSentenceWord.index, word: nonSentenceWord.word});
+        });
+
+        return sentenceWords;
+    }
+
+    // Case in-sensitive search if 'i' option provided.
+    function wordExistInDict(word, i) {
+        var result = {};
+        if (i) {
+            result = _.pick(dict.content, function(value, key) {
+                return key.toLowerCase() === word.toLowerCase();
+            });
+        } else {
+            result = _.pick(dict.content, function(value, key) {
+                return key === word;
+            });
+        }
+        return !_.isEmpty(result);
+    }
+
+    function verifyDictWord(word, index, objSentenceWord) {
+        if (index === 0) {
+            if (!isFirstLetterCapital(word)) {
+                return false;
+            } else {
+                return wordExistInDict(word, 'i');
+            }
+        } else {
+            if (!_.isEmpty(objSentenceWord)) {
+                if (isFirstLetterCapital(objSentenceWord[0].word)) {
+                    return wordExistInDict(word, 'i');
+                } else {
+                    return false;
+                }
+            }
+            return dict.content[word];
+        }
+    }
+
+    function isFirstLetterCapital(word) {
+        var re = /^[A-Z][a-z0-9_-]*$/;
```

Comment:
this should be probably only `[A-Z]`, given that it only checks first character.
it could be also `return word[0] === word[0].toUpperCase()` which should be faster